mlflow/entities/span_event.py

```diff
@@ -57,3 +60,30 @@ def _get_stacktrace(error: BaseException) -> str:
             return (msg + "\n\n".join(tb)).strip()
         except Exception:
             return msg
+
+    def json(self):
+        return {
+            "name": self.name,
+            "timestamp": self.timestamp,
+            "attributes": json.dumps(self.attributes, cls=CustomEncoder)
+            if self.attributes
+            else None,
+        }
+
+
+class CustomEncoder(json.JSONEncoder):
+    """
+    Custom encoder to handle json serialization.
+    """
+
+    def default(self, o):
+        # convert datetime to string format by default
+        if isinstance(o, datetime):
+            return o.isoformat()
+        if isinstance(o, uuid.UUID):
+            return str(o)
+        try:
+            return super().default(o)
+        # convert object direct to string to avoid error in serialization
+        except TypeError:
+            return str(o)
```

Comment:
```suggestion
        try:
            return super().default(o)
        # convert object direct to string to avoid error in serialization
        except TypeError:
            # convert datetime to string format by default
            if isinstance(o, datetime):
                return o.isoformat()
            if isinstance(o, uuid.UUID):
                return str(o)
            return str(o)
```

can we try `super().default` first to reduce useless `isinstance` calls?